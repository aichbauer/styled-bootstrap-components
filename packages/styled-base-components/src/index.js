import { BootstrapBaseCss } from './components';

export { BootstrapBaseCss };
