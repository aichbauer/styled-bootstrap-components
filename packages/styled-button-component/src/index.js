import {
  Button,
  LinkButton,
} from './components/Button';
import { ButtonGroup } from './components/ButtonGroup';
import { ButtonToolbar } from './components/ButtonToolbar';

export {
  Button,
  LinkButton,
  ButtonGroup,
  ButtonToolbar,
};
