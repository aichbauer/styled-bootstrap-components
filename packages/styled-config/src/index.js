import { theme } from './theme';
import {
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
} from './utils';

export {
  theme,
  // utils
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
};
