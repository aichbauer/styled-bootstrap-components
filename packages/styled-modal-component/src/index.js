import { Modal } from './components/Modal';
import { ModalWrapper } from './components/ModalWrapper';
import { ModalBody } from './components/ModalBody';
import { ModalContent } from './components/ModalContent';
import { ModalDialog } from './components/ModalDialog';
import { ModalFooter } from './components/ModalFooter';
import { ModalHeader } from './components/ModalHeader';
import { ModalTitle } from './components/ModalTitle';
import { ModalBackdrop } from './components/ModalBackdrop';

export {
  Modal,
  ModalWrapper,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  ModalBackdrop,
};
