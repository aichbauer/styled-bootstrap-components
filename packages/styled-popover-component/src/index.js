import { PopoverWithoutTransition, Popover } from './components/Popover';
import { PopoverArrow } from './components/PopoverArrow';
import { PopoverBody } from './components/PopoverBody';
import { PopoverHeader } from './components/PopoverHeader';

export {
  PopoverWithoutTransition,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
};
