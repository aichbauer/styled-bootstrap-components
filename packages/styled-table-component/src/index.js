import { Table } from './components/Table';
import { Tr } from './components/Tr';

export {
  Table,
  Tr,
};
