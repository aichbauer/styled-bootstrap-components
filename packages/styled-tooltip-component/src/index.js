import { TooltipWithoutTransition, Tooltip } from './components/Tooltip';
import { TooltipArrow } from './components/TooltipArrow';
import { TooltipInner } from './components/TooltipInner';

export {
  TooltipWithoutTransition,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
