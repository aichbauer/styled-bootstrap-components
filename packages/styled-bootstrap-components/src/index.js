import { Alert } from 'styled-alert-component';
import { Badge } from 'styled-badge-component';
import {
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  Button as BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Nav as Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  Table as BaseTable,
  Textarea,
  Th,
  Ul,
} from 'styled-base-components';
import {
  Breadcrumb,
  BreadcrumbItem,
} from 'styled-breadcrumb-component';
import {
  Button,
  ButtonGroup,
  ButtonToolbar,
} from 'styled-button-component';
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
} from 'styled-card-component';
import { screenSize } from 'styled-config';
import { Container } from 'styled-container-component';
import {
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
} from 'styled-dropdown-component';
import {
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
} from 'styled-form-component';
import {
  Column,
  Row,
} from 'styled-grid-system-component';
import { Jumbotron } from 'styled-jumbotron-component';
import {
  ListGroup,
  ListGroupItem,
} from 'styled-listgroup-component';
import {
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
} from 'styled-modal-component';
import {
  Nav,
  NavLink,
} from 'styled-nav-component';
import {
  Navbar,
  NavbarLink,
} from 'styled-navbar-component';
import {
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
} from 'styled-popover-component';
import {
  Table,
  Tr,
} from 'styled-table-component';
import {
  Tooltip,
  TooltipArrow,
  TooltipInner,
} from 'styled-tooltip-component';

export {
  Alert,
  Badge,
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  BaseTable,
  Textarea,
  Th,
  Ul,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  ButtonGroup,
  ButtonToolbar,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
  screenSize,
  Container,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
  Column,
  Row,
  Jumbotron,
  ListGroup,
  ListGroupItem,
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  Nav,
  NavLink,
  Navbar,
  NavbarLink,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
  Table,
  Tr,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
