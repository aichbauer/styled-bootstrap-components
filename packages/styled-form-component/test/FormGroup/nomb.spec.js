import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { FormGroup } from '../../src';

test('Styles matches FormGroup nomb', () => {
  const tree = renderer.create(<FormGroup nomb />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('margin-bottom', '0');
});

test('Styles matches FormGroup nomb row', () => {
  const tree = renderer.create(<FormGroup nomb row />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('margin-right', '-5px');
  expect(tree).toHaveStyleRule('margin-left', '-5px');
  expect(tree).toHaveStyleRule('margin-bottom', '0');
  expect(tree).toHaveStyleRule('margin-bottom', '0', {
    modifier: '& > div > label',
  });
});
