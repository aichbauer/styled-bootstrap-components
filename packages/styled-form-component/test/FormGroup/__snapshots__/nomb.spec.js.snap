// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles matches FormGroup nomb 1`] = `
.c0 {
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0 > input,
.c0 > div > input,
.c0 > :not(input[type=file]),
.c0 > div > :not(input[type=file]) {
  border-radius: 0.25rem;
}

.c0 > input,
.c0 > div > input,
.c0 > label,
.c0 > div > label {
  line-height: 1.5;
}

.c0 > input,
.c0 > div > input,
.c0 > label,
.c0 > div > label {
  font-size: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Styles matches FormGroup nomb row 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -5px;
  margin-left: -5px;
  margin-bottom: 0;
}

.c0 > div > label {
  margin-bottom: 0;
}

.c0 > input,
.c0 > div > input,
.c0 > :not(input[type=file]),
.c0 > div > :not(input[type=file]) {
  border-radius: 0.25rem;
}

.c0 > input,
.c0 > div > input,
.c0 > label,
.c0 > div > label {
  line-height: 1.5;
}

.c0 > input,
.c0 > div > input,
.c0 > label,
.c0 > div > label {
  font-size: 1rem;
}

<div
  className="c0"
/>
`;
